common/src/main/java/org/apache/falcon/workflow/WorkflowExecutionArgs.java

```diff
@@ -55,6 +55,7 @@
     STATUS("status", "status of the user workflow isnstance"),
     WF_ENGINE_URL("workflowEngineUrl", "url of workflow engine server, ex:oozie", false),
     USER_SUBFLOW_ID("subflowId", "external id of user workflow", false),
+    PARENT_ID("parentId", "The parent of the current workflow, typically coord action", false),
```

Comment:
Does it make sense for you to get the parent from workflow bean via oozie client on the re-run & late-re-run path instead of bouncing it of notification ?